torchvision/transforms/transforms.py

```diff
@@ -370,36 +395,57 @@ def __repr__(self):
             format(self.padding, self.fill, self.padding_mode)
 
 
-class Lambda(object):
+class Lambda(torch.nn.Module):
     """Apply a user-defined lambda as a transform.
 
+    .. Note::
+        This class exposes ``lambd`` attributes as its own attributes:
+
+        >>> # t.size = 10
+        >>> t_lambda = Lambda(t)
+        >>> assert t_lambda.size == t.size
+
     Args:
         lambd (function): Lambda/function to be used for transform.
     """
 
     def __init__(self, lambd):
-        assert callable(lambd), repr(type(lambd).__name__) + " object is not callable"
+        super().__init__()
+        if not callable(lambd):
+            raise TypeError("Argument lambd should be callable, got {}".format(repr(type(lambd).__name__)))
         self.lambd = lambd
 
-    def __call__(self, img):
+    def forward(self, img: Tensor) -> Tensor:
         return self.lambd(img)
 
     def __repr__(self):
         return self.__class__.__name__ + '()'
 
+    def __getattr__(self, attr):
+        if hasattr(self.lambd, attr):
+            return getattr(self.lambd, attr)
+        return super().__getattr__(attr)
```

Comment:
Hum... can you explain why we need to override `__getattr__`   and `__setattr__` here?